services/web/server/src/simcore_service_webserver/exporter/archiving.py

```diff
@@ -0,0 +1,123 @@
+import asyncio
+import logging
+from pathlib import Path
+from typing import List, Tuple
+
+from passlib import pwd
+
+from .async_hashing import Algorithm, checksum
+from .exceptions import ExporterException
+from .utils import rename
+
+log = logging.getLogger(__name__)
+
+
+def get_random_chars(length: int) -> str:
+    return pwd.genword(entropy=52, charset="hex")[:length]
+
+
+def get_osparc_export_name(sha256_sum: str, algorithm: Algorithm) -> str:
+    return f"{get_random_chars(4)}#{algorithm.name}={sha256_sum}.osparc"
+
+
+def validate_osparc_import_name(file_name: str) -> Tuple[Algorithm, str]:
+    """returns: sha256 from file signature if present or raises an error"""
+    # sample: 0a3a#SHA256=02d03b65911aae7662a8fa5fa4847d0b8f722a022d95b37b4a02960ff736853a.osparc
+    if not file_name.endswith(".osparc"):
+        raise ExporterException(
+            f"Provided file name must haave .osparc extension file_name={file_name}"
+        )
+
+    parts = file_name.replace(".osparc", "").split("#")
+    if len(parts) != 2:
+        raise ExporterException(
+            f"Could not find a digest in provided file_name={file_name}"
+        )
+    digest = parts[1]
+
+    digest_parts = digest.split("=")
+    if len(digest_parts) != 2:
+        raise ExporterException(
+            f"Could not find a valid digest in provided file_name={file_name}"
+        )
+    algorithm, digest_sum = Algorithm[digest_parts[0]], digest_parts[1]
+    return algorithm, digest_sum
+
+
+def search_for_unzipped_path(search_path: Path) -> Path:
+    found_dirs = []
+    for path in search_path.iterdir():
+        if path.is_dir():
+            found_dirs.append(path)
+
+    if len(found_dirs) != 1:
+        raise ExporterException(
+            f"unexpected number of directories after unzipping {found_dirs}"
+        )
+    return search_path / found_dirs[0]
+
+
+async def _wrap_create_subprocess_exec(
+    command_args: List[str], cwd_path: Path, fail_message: str
+) -> None:
+    proc = await asyncio.create_subprocess_exec(
+        *command_args,
+        cwd=str(cwd_path),
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+    )
+    stdout, stderr = await proc.communicate()
+
+    if proc.returncode != 0:
+        log.warning("STDOUT: %s", stdout.decode())
+        log.warning("STDERR: %s", stderr.decode())
+        raise ExporterException(fail_message)
+
+
+async def zip_folder(project_id: str, input_path: Path, no_compression=False) -> Path:
+    """Zips a folder and returns the path to the new archive"""
+
+    zip_file = Path(input_path.parent) / f"{project_id}.zip"
+    if zip_file.is_file():
+        raise ExporterException(
+            f"Cannot archive because file already exists '{str(zip_file)}'"
+        )
+
+    command_args = [
+        "zip",
```

Comment:
remind me why you use command line utility instead of built-in library ``zipfile``? 
SEE https://www.peterbe.com/plog/fastest-way-to-unzip-a-zip-file-in-python

Reply:
Because that was the fastest solution to code which did not block the aioloop.

Reply:
> Because that was the fastest that came to mind which **did not cause the loop to block**

SEE this example https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor

Reply:
done, migrated